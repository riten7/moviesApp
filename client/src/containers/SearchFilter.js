import React, { useCallback } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setSearchText, showMoviePopup } from '../actions/actionCreators';
import Popup from './Popup';
import { Col, Input, Button } from 'antd';
import { PlusOutlined } from '@ant-design/icons';

const SearchFilter = () => {
  const dispatch = useDispatch();
  const moviePopupShown = useSelector(state => state.popupShown);

  const handleInputChange = useCallback((e) => {
    dispatch(setSearchText(e.target.value));
  }, [dispatch]);

  const handleAddMovieBtn = useCallback(() => dispatch(showMoviePopup(true)), [dispatch]);

  return (
    <>
    <div className="search-form">
      <Col>
        <Input className="searchInput" type="text" placeholder="Search your movie here..."
          onChange={handleInputChange} />
          <p>Search your favourite movies by typing.</p>
      </Col>
      <Col>
        <Button className="addMovieBtn" type="primary" onClick={handleAddMovieBtn}><PlusOutlined /></Button>
      </Col>
      </div>
      {moviePopupShown ? <Popup /> : null}
    </>
  );
}

export default SearchFilter;