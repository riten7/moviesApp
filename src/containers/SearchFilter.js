import React, { useCallback, useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { setSearchText, addMovieToList } from '../actions/actionCreators';
import MoviePopup from './MoviePopup';
import { Col, Input, Button, Spin } from 'antd';
import { PlusOutlined } from '@ant-design/icons';

const SearchFilter = () => {
  const [popupDisplay, setPopupDisplay] = useState(false);
  const [loader, setLoader] = useState(false);
  const dispatch = useDispatch();
  const { status } = useSelector(state => state.movieList);

  useEffect(() => {
    if (status === 'completed') {
      setPopupDisplay(false);
    }
  }, [status]);

  const handleInputChange = useCallback((e) => {
    dispatch(setSearchText(e.target.value));
  }, [dispatch]);

  const submitMovie = (values) => {
    setLoader(true);
    dispatch(addMovieToList(values));
  }

  const setPopupState = (value) => setPopupDisplay(value);

  return (
    <>
      <div className="search-form">
        <Col>
          <Input className="searchInput" type="text" placeholder="Search your movie here..."
            onChange={handleInputChange} />
          <p>Search your favourite movies by typing.</p>
        </Col>
        <Col>
          <Button className="addMovieBtn" type="primary" onClick={setPopupState}><PlusOutlined /></Button>
        </Col>
      </div>
      {popupDisplay ?<MoviePopup
        submitMovie={submitMovie}
        setPopupState={setPopupState} />: null}
    </>
  );
}

export default SearchFilter;