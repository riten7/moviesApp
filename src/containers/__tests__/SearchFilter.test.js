import React from "react";
import { mount } from "enzyme";
import SearchFilter from "../SearchFilter";
import configureStore from "redux-mock-store";
import thunk from 'redux-thunk';
import * as ReactReduxHooks from "../react-redux-hooks";
import {Provider} from 'react-redux';
import data from '../__mockData__/movielistData.json';
import movies from '../__mockData__/movies.json';

window.matchMedia = jest.fn().mockImplementation((query) => ({
  matches: false,
  media: query,
  onchange: null,
  addListener: jest.fn(), // deprecated
  removeListener: jest.fn(), // deprecated
  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
  dispatchEvent: jest.fn(),
}))

describe("Component renders properly", () => {
  let store;
  let wrapper;
   beforeEach(() => {
    /* mocking store */
    store = configureStore([thunk])(data);

    /* mocking useSelector on our mock store */
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation(state => store.getState());

    /* mocking useDispatch on our mock store  */
    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);  /* shallow rendering */
  });

  it("Test SearchFilter Component works", () => {
    wrapper = mount(<Provider store={store}><SearchFilter/></Provider>);
    expect(wrapper.html()).toMatchSnapshot();

    wrapper.find('.addMovieBtn').at(0).simulate("click");
    expect(store.getState().movieList.status).toEqual('completed');
    expect(store.getState().movieList.movies).toEqual(movies);

    wrapper.find('.searchInput').at(0).simulate('change', { target: {value: 'MR' }});
    expect(store.getActions()).toEqual( [{"payload": {"searchtext": "MR"}, "type": "SEARCH_FILTER"}] );
  });
});