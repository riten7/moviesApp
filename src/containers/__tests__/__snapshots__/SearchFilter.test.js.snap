// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders properly Test SearchFilter Component works 1`] = `"<div class=\\"search-form\\"><div class=\\"ant-col\\"><input class=\\"ant-input searchInput\\" type=\\"text\\" placeholder=\\"Search your movie here...\\" value=\\"\\"><p>Search your favourite movies by typing.</p></div><div class=\\"ant-col\\"><button type=\\"button\\" class=\\"ant-btn addMovieBtn ant-btn-primary\\"><span role=\\"img\\" aria-label=\\"plus\\" class=\\"anticon anticon-plus\\"><svg viewBox=\\"64 64 896 896\\" focusable=\\"false\\" class=\\"\\" data-icon=\\"plus\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\"><defs><style></style></defs><path d=\\"M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z\\"></path><path d=\\"M176 474h672q8 0 8 8v60q0 8-8 8H176q-8 0-8-8v-60q0-8 8-8z\\"></path></svg></span></button><p>Add Movie</p></div></div>"`;
